pilot/pkg/networking/util/util.go

```diff
@@ -616,8 +610,12 @@ func BuildLbEndpointMetadata(uid string, network string, mtlsReady bool) *core.M
 		}
 	}
 
-	if mtlsReady {
-		metadata.FilterMetadata[EnvoyTransportSocketMetadataKey] = EndpointMetadataMtlsReady
+	if tlsMode != "" {
+		metadata.FilterMetadata[EnvoyTransportSocketMetadataKey] = &pstruct.Struct{
```

Comment:
@howardjohn raised a performance concern with initializing this struct/map constantly [see here](https://github.com/istio/istio/pull/17621#discussion_r335547832), but I guess this can't be avoided since the key's value is dynamic rather than always `true`?

Reply:
Yes. And the goal of this is to allow end users to provide any value for the key and not just the specific values. By allowing them to provide any value, they can then use this key-value to create otehr custom transport socket matchers (using EnvoyFilter). Essentially, in this case, we are trading off performance for flexibility.

Reply:
I think its fine, if we really need to we can add a cache per tlsMode, but probably premature optimization